addons/mail/static/src/new/core/chat_window_model.js

```diff
@@ -32,8 +35,25 @@ export class ChatWindow {
             _state: state,
         });
         this.update(data);
-        state.chatWindows.push(this);
-        return state.chatWindows.find((c) => c.threadLocalId === data.threadLocalId); // return reactive version
+        let idx;
+        if (!data.replaceNewMessageChatWindow) {
+            idx = state.chatWindows.length;
+        } else {
+            const newMessageChatWindowIndex = state.chatWindows.findIndex(
+                ({ threadLocalId }) => !threadLocalId
+            );
+            idx = newMessageChatWindowIndex !== -1 ? newMessageChatWindowIndex : state.chatWindows.length;
+        }
+        state.chatWindows.splice(idx, 1, this);
+        return state.chatWindows[idx]; // return reactive version
```

Comment:
Normally all this code should not be in `constructor`, but in `static insert`.
It's weird that the `constructor` returns another chat window.